modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/SelectInsertStoreQueryRuntime.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.siddhi.core.query;
+
+import org.wso2.siddhi.core.event.ComplexEvent;
+import org.wso2.siddhi.core.event.ComplexEventChunk;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.event.state.StateEvent;
+import org.wso2.siddhi.core.event.state.StateEventPool;
+import org.wso2.siddhi.core.event.stream.MetaStreamEvent;
+import org.wso2.siddhi.core.event.stream.StreamEvent;
+import org.wso2.siddhi.core.exception.StoreQueryRuntimeException;
+import org.wso2.siddhi.core.query.selector.QuerySelector;
+import org.wso2.siddhi.query.api.definition.Attribute;
+
+import java.util.Arrays;
+import java.util.List;
+
+
+/**
+ * This class is used to keep the runtime information needed to execute a select insert store query.
+ */
+public class SelectInsertStoreQueryRuntime implements StoreQueryRuntime {
+
+    private String queryName;
+    private MetaStreamEvent.EventType eventType;
+    private QuerySelector selector;
+    private StateEventPool stateEventPool;
+    private MetaStreamEvent metaStreamEvent;
+    private Attribute[] outputAttributes;
+
+    public SelectInsertStoreQueryRuntime(String queryName, MetaStreamEvent metaStreamEvent) {
+        this.queryName = queryName;
+        this.eventType = metaStreamEvent.getEventType();
+        this.metaStreamEvent = metaStreamEvent;
+        this.setOutputAttributes(metaStreamEvent.getLastInputDefinition().getAttributeList());
+    }
+
+    @Override
+    public Event[] execute() {
+        try {
+            StateEvent stateEvent = new StateEvent(1, outputAttributes.length);
+            StreamEvent streamEvent = new StreamEvent(metaStreamEvent.getBeforeWindowData().size(),
+                    metaStreamEvent.getOnAfterWindowData().size(),
+                    metaStreamEvent.getOutputData().size());
+            stateEvent.addEvent(0, streamEvent);
+
+            ComplexEventChunk complexEventChunk = new ComplexEventChunk(stateEvent, stateEvent, true);
+
+            if (eventType == MetaStreamEvent.EventType.TABLE) {
```

Comment:
Will this be any other value other than EventType.TABLE, then we have to throw an exception!

Reply:
Fixed with e206a50